Simplified/NYPLNetworkQueue.swift

```diff
@@ -0,0 +1,173 @@
+import Foundation
+import SQLite
+
+let OfflineQueueStatusCodes = [NSURLErrorTimedOut,
+                               NSURLErrorCannotFindHost,
+                               NSURLErrorCannotConnectToHost,
+                               NSURLErrorNetworkConnectionLost,
+                               NSURLErrorNotConnectedToInternet,
+                               NSURLErrorInternationalRoamingOff,
+                               NSURLErrorCallIsActive,
+                               NSURLErrorDataNotAllowed,
+                               NSURLErrorSecureConnectionFailed]
+let QueueMaxRetryCount = 3
+
+enum HTTPMethodType: String {
+  case GET, POST, HEAD, PUT, DELETE, OPTIONS, CONNECT
+}
+
+final class NetworkQueue: NSObject {
+  
+  private static let path = NSSearchPathForDirectoriesInDomains(.applicationSupportDirectory, .userDomainMask, true).first!
+  
+  private static let sqlTable = Table("queueTable")
+  
+  private static let sqlID = Expression<Int>("rowid")
+  private static let sqlLibraryID = Expression<Int>("libraryIdentifier")
+  private static let sqlUpdateID = Expression<String?>("updateIdentifier")
+  private static let sqlUrl = Expression<String>("requestUrl")
+  private static let sqlMethod = Expression<String>("requestMethod")
+  private static let sqlParameters = Expression<Data?>("requestParameters")
+  private static let sqlHeader = Expression<Data?>("requestHeader")
+  private static let sqlRetries = Expression<Int>("retryCount")
+  
+  
+  // MARK: - Public Functions
+  
+  class func addRequest(_ libraryID: Int,
```

Comment:
I think they're both risks, yes! I suspect is more likely that a PIN will change that an account will be closed though for whatever that's worth. We should be able to reasonably handle both cases regardless.